File: Turret-Shot/Turret Shot/GameMechanics/Player.swift
//
//  Player.swift
//  Turret Shot
//
//  Created by 90307297 on 2/17/22.
//

import Foundation

class Player:MovingObject{
    var aliveStatus: Bool //True if the player is alive
    var shotStatus: Bool  //True if it shot
    var moveStatus: Bool  //True if it moved
    var playerNum: Int    //Player label
    var shots: [Shot]
    
    init(xP: Int, yP: Int, xO: Int, yO: Int, n: Int){
        self.playerNum = n
        self.aliveStatus = true
        self.moveStatus = true
        self.shotStatus = true
        
        shots = []
        
        super.init(xP: xP, yP: yP, xO: xO, yO: yP)
    }
    
    
}

File: Turret-Shot/Player.java
import java.util.ArrayList;

public class Player extends MovingObject{

  private boolean aliveStatus; //True if the player is alive
  private boolean shotStatus; //True if it shot
  private boolean moveStatus; //True if it moved
  private int playerNum; //Player label
  private ArrayList<Shot> shots;

  public Player(int xP, int yP, int xO, int yO, int n){
    super(xP, yP, xO, yO);
    this.aliveStatus = true;
    this.shotStatus = false;
    this.moveStatus = false;
    this.playerNum = n;
    shots = new ArrayList<>();
  }

  public void makeShot(Player o){
    shots.add(new Shot(o));
  }

  public boolean getAliveStatus(){
    return aliveStatus;
  }

  public boolean getShotStatus(){
    return shotStatus;
  }

  public boolean getMoveStatus(){
    return moveStatus;
  }
  
  public int getPlayerNum(){
    return playerNum;
  }

  public ArrayList<Shot> getShots(){
    return shots;
  }

  public void setAliveStatus(boolean stat){
    this.aliveStatus = stat;
  }

  public void setShotStatus(boolean stat){
    this.shotStatus = stat;
  }

  public void setMoveStatus(boolean stat){
    this.moveStatus = stat;
  }
}